kong/dao/schemas/upstreams.lua

```diff
@@ -238,6 +248,14 @@ return {
                                   "but no header name provided")
     end
 
+    if (config.hash_on == "cookie"
+        and not config.hash_on_cookie) or
+       (config.hash_fallback == "cookie"
+        and not config.hash_fallback_cookie) then
+      return false, Errors.schema("Hashing on 'cookie', " ..
+                                  "but no cookie name provided")
+    end
+
```

Comment:
also relevant here: if the primary is set to "cookie" then the fallback must be "none". Because if the cookie isn't set, it will be created, hence the fallback will never be invoked in that case.